W-16161768 upload to google drive (#413)
mulesoft/docs-rpa: rpa-builder/6.7/modules/ROOT/pages/toolbox-google-workspace-upload-to-google-drive.adoc

= Upload to Google Drive 

> Upload files to your Google Drive. 

Reviewer: s/d: summarize the topic and give user value.

> You can export Google Docs, Sheets, and Slides into a selection of supported formats including PDF.

Reviewer 2: This sentence can be removed, think it is still from the Download from Google Drive

== Before You Begin 

> Before uploading a file, ensure you: 

Reviewer: See previous suggestion to make these imperative/present tense.

* Configured your Google API Service Account Credentials.
+
You can configure your credentials as an xref:toolbox-variable-handling-activity-parameters.adoc#defining-activity-parameters[Activity Parameter] or inline in the action step wizard. 

== Properties 
//OAuth common properties
include::partial$oauth-authentication-settings.adoc[]

* *Common* 
+

> See xref:common-properties.adoc[Common Properties]

Reviewer: ```suggestion
See xref:common-properties.adoc[Common Properties].
```

Reviewer: Hard-coded link text?

> * *File Operation Settings*

Reviewer: See prev suggestion about using a table/sentence fragments don't get periods.

** *Directory path* 
+
Path to the directory where the file to upload is. 
+

> You can specify either an absolute path or a relative to the Base Path. 

Reviewer: ```suggestion
You can specify an absolute path or a path relative to the Base Path. 
```

** *File name*
+
Name of the file to upload. 
** *Google Drive Folder ID*
+
ID of the destination Google Drive folder where to upload the file. 
+

> You can obtain the folder ID by navigating to said Google Drive folder and extracting the ID from the URL (the ID is the value `folder/`). If you don't specify a folder ID, this action step upload the file to the root folder. 

Reviewer 2: We should add here that users can also obtain the folder ID by using the Search in Google Drive action step and "Search for: Folders" there, they will get Folder IDs matching their search criteria.

> You can obtain the folder ID by using the Search in Google Drive action step and searching for the folder, or by navigating to said Google Drive folder and extracting the ID from the URL (the ID is the value `folder/`). If you don't specify a folder ID, this action step uploads the file to the root folder. 

Reviewer: ```suggestion
To get the folder ID, use Search in the Google Drive action step to find the folder. You can also go to the Google Drive folder and extract the ID (the value `folder/`) from the URL. If you don't specify a folder ID, this action step uploads the file to the root folder. 
```

== Inbound Variables 

* *Directory path*
* *File name*
* *Google Drive Folder ID*
//OAuth common inbound variables
include::partial$oauth-inbound-variables.adoc[]

== Outbound Variables

> * *Upload successful* (Boolean)

Reviewer: See previous comment on similar boolean variable. 

No terminating punctuation for sentence fragments.

+
Confirms if the upload was successful.
* *File ID in Google Drive*
+
Returns the Google Drive file ID of the uploaded file.

> == Wizard 

Reviewer: See prev comment about making this the task (imperative) and moving to the beginning of the file.

include::partial$oauth-authentication-wizard.adoc[tag=authentication-info]

> Specify the *Directory path* and the *File name* of the file to upload. You can provide a *Google Drive Folder ID* to upload the file into this folder. If you don't provide this value, this action step uploads the file to the root folder of your Google Drive. 

Reviewer: To avoid using UI elements as words in the sentence, lowercase and remove bold.
